src/PickerAvoidingView/index.ios.js

```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Animated, StyleSheet, View } from 'react-native';
+import { PickerStateContext } from '../PickerStateProvider';
+import { IOS_MODAL_ANIMATION_DURATION_MS, IOS_MODAL_HEIGHT } from '../constants';
+
+const styles = {
+    container: {
+        flexDirection: 'column',
+        justifyContent: 'flex-start',
+        alignItems: 'stretch',
+    },
+    contentContainer: {
+        flex: 1,
+    },
+};
+
+/**
+ * PickerAvoidingView is a React component that adjusts the view layout to avoid
+ * being covered by an open iOS UIPickerView modal. It's meant to be similar to
+ * the built-in KeyboardAvoidingView component, but specifically tailored for
+ * iOS picker modals.
+ *
+ * In order for this component to work correctly, all the pickers and the
+ * PickerAvoidingView should have a PickerStateProvider ancestor.
+ *
+ * @param {React.ReactNode} props.children - The child components that should be
+ * protected from obstruction by the picker modal
+ */
+export function PickerAvoidingView(props) {
+    const context = React.useContext(PickerStateContext);
+    const isPickerOpen = context && context.isPickerOpen;
+
+    const spaceHeight = React.useRef(new Animated.Value(0)).current;
+
+    React.useEffect(() => {
+        if (isPickerOpen) {
+            Animated.timing(spaceHeight, {
```

Comment:
When I started implementing this, I thought that I actually needed it to be animated. But now, when I'm looking at this, it seems that it basically regressed to a `setTimeout`. I'll try simplifying it.

Reply:
Done